README.md

```diff
@@ -1 +1,94 @@
-Placeholder for Checkstyle Regression Tool readme.
+# Regression-tool
+
+Regression-tool is an automation tool to do regression testing based on proposed patch(Pull Request) 
+of checkstyle project. The tool could generate configuration based on Git changes, which is used to 
+generate diff report between the base and the patch branch.
+
+## Setup
+
+### Requirements
+
+- Checkstyle repository which including both master and patch branch need to be cloned.
+In most case, it should be your forked repository.
+- [Contribution repository](https://github.com/checkstyle/contribution) need to be cloned.
+- Regression-tool uses environment variable `M2_HOME` or system property `maven.home` to find the maven binary files.
+You need to specify either `M2_HOME` or `maven.home` in advance. You could get the path of maven home by running `mvn -v`.
+
+### Clone
+
+Just clone this repository to your local.
+
+```bash
+$ git clone git@github.com:checkstyle/regression-tool.git
+```
+
+Or use HTTPS mode.
+```bash
+$ git clone https://github.com/checkstyle/regression-tool.git
+```
+
+### Generate Jar
+
+Generate a `regression-tool-XX-all.jar`(XX means the version of regression-tool) in the `target` folder.
+
+```bash
+$ cd /path/to/regression-tool
+$ mvn clean package -Passembly
+```
+
+## Usage
+
+```bash
+$ java -jar regression-tool-XX-all.jar -p <arg> -r <arg> [--stopAfterGenerateConfig] [-t <arg>]
+```
+
+### Arguments
+
+#### localGitRepo (r)
```

Comment:
this is bad name, as it is not clear what repo is meant: checkstyle, regression-tool, contribtions ?